test/common/common/utility_test.cc

```diff
@@ -1095,6 +1095,60 @@ TEST(InlineStorageTest, InlineString) {
   EXPECT_EQ("Hello, world!", hello->toString());
 }
 
+TEST(TinyStringTest, Empty) {
```

Comment:
I think a sizeof test that demonstrates the win over `absl::optional<std::string>` would be beneficial, similar to the one you have in PR #24661

Reply:
makes sense.